### PR TITLE
CI: Refactor K8s test to use `kubectl get cep` 

### PR DESCRIPTION
**Pull request by eloycoto**
I made a refactor on test: 

- Use always `kubectl get cep` instead of executing commands on cilium pods. 
- Remove old functions that are not longer needed. 

Each commit explain what I did.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4640)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by tgraf** [comment]
Please provide a more descriptive PR title which indicates that this is CI work and what it rewrites.

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -87,6 +86,38 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 			appPods   map[string]string
 		)
 
+		validateNoPolicy := func() {
+
+			ExpectWithOffset(1, kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range apps {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				ExpectWithOffset(1, cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exits --> exist

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -87,6 +86,38 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+					"Policy status does not match for endpoint %s", pod)
+			}
+		}
+
+		validatePolicyStatus := func() {
+			ExpectWithOffset(1, kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App2, helpers.App3} {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				ExpectWithOffset(1, cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exits --> exist

File: test/k8sT/Policies.go [hunk 3]
```diff
@@ -87,6 +86,38 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 			appPods   map[string]string
 		)
 
+		validateNoPolicy := func() {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: validateNoPolicy --> validateNoPolicyEnabled

File: test/k8sT/Policies.go [hunk 4]
```diff
@@ -87,6 +86,38 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				ExpectWithOffset(1, cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
+
+				ExpectWithOffset(1, cep.Status.Policy.Realized.PolicyEnabled).To(
+					Equal(models.EndpointPolicyEnabledNone),
+					"Policy status does not match for endpoint %s", pod)
+			}
+		}
+
+		validatePolicyStatus := func() {
```
Comment (ianvernon) [inline test/k8sT/Policies.go:104]: Add small comments documenting what this function does; i.e., tests that app2 and app3 have no policy enforcement enabled, and that app1 has ingress policy enforcement enabled.

File: test/k8sT/Policies.go (hunk 3 above)
Comment (ianvernon) [inline test/k8sT/Policies.go]: Add a small comment indicating what this function does.

File: test/k8sT/Policies.go [hunk 5]
```diff
@@ -139,56 +170,43 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-			Expect(policyStatus[models.EndpointPolicyEnabledIngress]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledEgress]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledBoth]).Should(Equal(4))
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
 
-			By("Return PolicyEnforcement to default")
+			for _, app := range apps {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				Expect(cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exits --> exist

File: test/k8sT/Policies.go [hunk 6]
```diff
@@ -139,56 +170,43 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-			policyStatus = endpoints.GetPolicyStatus()
-			// Always-on mode with no policy, all endpoints must be in default deny
-			Expect(policyStatus[models.EndpointPolicyEnabledNone]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledIngress]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledEgress]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledBoth]).Should(Equal(4))
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
 
-			By("Return PolicyEnforcement to default")
+			for _, app := range apps {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Add a small comment indicating what is being checked, e.g. 
```
// check that policy enforcement is enabled for all app pods
```

File: test/k8sT/Policies.go [hunk 7]
```diff
@@ -381,22 +385,17 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-			for _, ep := range epList {
-				for _, lbls := range ep.Status.Labels.SecurityRelevant {
-					if lbls == "k8s:io.kubernetes.pod.namespace="+helpers.DefaultNamespace {
-						switch ep.Status.Policy.Realized.PolicyEnabled {
-						case models.EndpointPolicyEnabledBoth, models.EndpointPolicyEnabledEgress:
-							epsWithEgress++
-						}
-					}
-				}
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Nit: in a timeout --> after a  specified timeout

Change this for all CEP-related timeout error description messages in this commit.

File: test/k8sT/Policies.go [hunk 8]
```diff
@@ -364,6 +357,17 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 				helpers.DefaultNamespace, appPods[helpers.App3],
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
 			res.ExpectFail("Ingress connectivity should be denied by policy")
+
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				Expect(cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exits --> exist

File: test/k8sT/Policies.go [hunk 9]
```diff
@@ -364,6 +357,17 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 				helpers.DefaultNamespace, appPods[helpers.App3],
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
 			res.ExpectFail("Ingress connectivity should be denied by policy")
+
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Instead of this slice, use the `apps` slice

Comment (ianvernon) [inline test/k8sT/Policies.go]: Add a `By` indicating that we are checking that ingress policy enforcement is enabled on all pods

File: test/k8sT/Policies.go [hunk 10]
```diff
@@ -381,22 +385,17 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-						case models.EndpointPolicyEnabledBoth, models.EndpointPolicyEnabledEgress:
-							epsWithEgress++
-						}
-					}
-				}
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				Expect(cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exits --> exist

File: test/k8sT/Policies.go [hunk 11]
```diff
@@ -381,22 +385,17 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-				for _, lbls := range ep.Status.Labels.SecurityRelevant {
-					if lbls == "k8s:io.kubernetes.pod.namespace="+helpers.DefaultNamespace {
-						switch ep.Status.Policy.Realized.PolicyEnabled {
-						case models.EndpointPolicyEnabledBoth, models.EndpointPolicyEnabledEgress:
-							epsWithEgress++
-						}
-					}
-				}
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Use apps slice

File: test/k8sT/Policies.go [hunk 12]
```diff
@@ -430,22 +429,17 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Use apps slice

File: test/k8sT/Policies.go [hunk 13]
```diff
@@ -478,68 +471,59 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 
 		It("Denies traffic with cnp default-deny ingress policy", func() {
 
-			res := kubectl.ExecPodCmd(
```
Comment (ianvernon) [inline test/k8sT/Policies.go:481]: Why was this removed? I see no reason to remove the checking of connectivity before importing policy.

File: test/k8sT/Policies.go [hunk 14]
```diff
@@ -478,68 +471,59 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-
 			By("Installing cnp ingress default-deny")
 
 			_, err := kubectl.CiliumPolicyAction(
 				helpers.KubeSystemNamespace, cnpDenyIngress, helpers.KubectlApply, 300)
 			Expect(err).Should(BeNil(),
 				"L3 deny-ingress Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
 
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: use apps slice

File: test/k8sT/Policies.go [hunk 15]
```diff
@@ -478,68 +471,59 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
 			res.ExpectFail("Ingress connectivity should be denied by policy")
 
+			By("Testing egress connnectivity works correctly")
 			res = kubectl.ExecPodCmd(
-				helpers.DefaultNamespace, appPods[helpers.App3],
-				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			res.ExpectFail("Ingress connectivity should be denied by policy")
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.Ping("8.8.8.8"))
```
Comment (ianvernon) [inline test/k8sT/Policies.go:506]: Why the move from curl to ping?

File: test/k8sT/Policies.go [hunk 16]
```diff
@@ -478,68 +471,59 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-				for _, lbls := range ep.Status.Labels.SecurityRelevant {
-					if lbls == "k8s:io.kubernetes.pod.namespace="+helpers.DefaultNamespace {
-						switch ep.Status.Policy.Realized.PolicyEnabled {
-						case models.EndpointPolicyEnabledBoth, models.EndpointPolicyEnabledEgress:
-							epsWithEgress++
-						}
-					}
-				}
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: use apps slice

File: test/k8sT/Policies.go [hunk 17]
```diff
@@ -478,68 +471,59 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				Expect(cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exits --> exist

File: test/k8sT/Policies.go [hunk 18]
```diff
@@ -430,22 +429,17 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 
 			By("Testing if egress policy enforcement is enabled on the endpoint")
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: This `By` is actually incorrect; we should update it to say that we are ensuring that both ingress and egress policy are being enforced.

File: test/k8sT/Policies.go [hunk 19]
```diff
@@ -566,18 +550,18 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-			Expect(err).Should(BeNil())
-			Expect(endpoints.AreReady()).Should(BeTrue())
-			policyStatus := endpoints.GetPolicyStatus()
-			// Always-on mode with no policy, all endpoints must be in default deny
-			Expect(policyStatus[models.EndpointPolicyEnabledNone]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledIngress]).Should(Equal(4))
-			Expect(policyStatus[models.EndpointPolicyEnabledEgress]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledBoth]).Should(Equal(0))
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Use apps slice

File: test/k8sT/Policies.go [hunk 20]
```diff
@@ -566,18 +550,18 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				Expect(cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exits --> exist

File: test/k8sT/Policies.go [hunk 21]
```diff
@@ -603,16 +587,16 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-			Expect(err).Should(BeNil())
-			Expect(endpoints.AreReady()).Should(BeTrue())
-			policyStatus := endpoints.GetPolicyStatus()
-			// Always-on mode with no policy, all endpoints must be in default deny
-			Expect(policyStatus[models.EndpointPolicyEnabledNone]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledIngress]).Should(Equal(0))
-			Expect(policyStatus[models.EndpointPolicyEnabledEgress]).Should(Equal(4))
-			Expect(policyStatus[models.EndpointPolicyEnabledBoth]).Should(Equal(0))
 
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: use apps slice

File: test/k8sT/Services.go [hunk 22]
```diff
@@ -205,81 +205,45 @@ var _ = Describe("K8sValidatedServicesTest", func() {
...
 		BeforeEach(func() {
-			servicePath = helpers.ManifestGet("external_service.yaml")
-			res := kubectl.Apply(servicePath)
-			res.ExpectSuccess(res.GetDebugMessage())
-
-			endpointPath = helpers.ManifestGet("external_endpoint.yaml")
-			podPath = helpers.ManifestGet("external_pod.yaml")
-			policyPath = helpers.ManifestGet("external_policy.yaml")
-			policyLabeledPath = helpers.ManifestGet("external_policy_labeled.yaml")
+			kubectl.Apply(servicePath).ExpectSuccess("cannot install external service")
```
Comment (ianvernon) [inline test/k8sT/Services.go:219]: IMO i think we should still error check any `Apply` statements that fail in before each.

File: test/k8sT/demos.go [hunk 23]
```diff
@@ -92,43 +92,28 @@ var _ = Describe(demoTestName, func() {
 	It("Tests Star Wars Demo", func() {
 
 		allianceLabel := "org=alliance"
-		empireLabel := "org=empire"
+		// empireLabel := "org=empire"
```
Comment (ianvernon) [inline test/k8sT/demos.go]: Remove this if the label is no longer used.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Love this change overall! Just some nitpicks / needs for more documentation within the tests about what is being tested for maintainability.

File: test/k8sT/Services.go (hunk 22 above)
Comment (ianvernon) [reply test/k8sT/Services.go:219]: Edit: ignore this, I am wrong :)

File: test/k8sT/Policies.go (hunk 15 above)
Comment (eloycoto) [reply test/k8sT/Policies.go:506]: Because we tested from APP2/APP3 to connect to clusterIP (app1-service) but we don't test any egress to world connectivity.

File: test/k8sT/Policies.go (hunk 4 above)
Comment (eloycoto) [reply test/k8sT/Policies.go:104]: done

File: test/k8sT/Policies.go (hunk 13 above)
Comment (eloycoto) [reply test/k8sT/Policies.go:481]: We already test in the first test that the connectivity works correctly. In all policy test we make the policy import and we don't test the connectivity. The main reason, is to keep the test as small as possible.

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/k8sT/Policies.go [hunk 24]
```diff
@@ -87,6 +86,42 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 		)
 
+		validateNoPolicyEnabled := func() {
+			// Validates that the `apps` pods do not have any policy
+			// enforcement enabled.
+			ExpectWithOffset(1, kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready after a specified timeout")
+			for _, app := range apps {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				ExpectWithOffset(1, cep).NotTo(BeNil(), "Endpoint %q does not exit", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exist

File: test/k8sT/Policies.go [hunk 25]
```diff
@@ -87,6 +86,42 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+
+		validatePolicyStatus := func() {
+			// Validates that app2 and app3 do not have a policy enforcement
+			// enabled. And validate that the app1 pods have ingress policy
+			// loaded.
+			ExpectWithOffset(1, kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready after a specified timeout")
+			for _, app := range []string{helpers.App2, helpers.App3} {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				ExpectWithOffset(1, cep).NotTo(BeNil(), "Endpoint %q does not exit", pod)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: exit --> exist. Please update accordingly for all other error messages.

File: test/k8sT/Policies.go [hunk 26]
```diff
@@ -478,69 +475,60 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-				for _, lbls := range ep.Status.Labels.SecurityRelevant {
-					if lbls == "k8s:io.kubernetes.pod.namespace="+helpers.DefaultNamespace {
-						switch ep.Status.Policy.Realized.PolicyEnabled {
-						case models.EndpointPolicyEnabledBoth, models.EndpointPolicyEnabledEgress:
-							epsWithEgress++
-						}
-					}
-				}
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready after a specified timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: use apps slice

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
